### PR TITLE
Update Grover tutorials to use Sampler

### LINKED ISSUE #1365: Update 06_grover.ipynb and 07_grover_examples.ipynb
### What is the expected behavior?

Resolve the following problems in https://github.com/Qiskit/qiskit-tutorials/blob/master/tutorials/algorithms/06_grover.ipynb and https://github.com/Qiskit/qiskit-tutorials/blob/master/tutorials/algorithms/07_grover_examples.ipynb:

- Uses quantum instance, opflow in examples,
- Uses old Grover and also addresses old amplitude amplification.

### PR DESCRIPTION
**Pull request by a-matsuo**
### Summary
This PR updates Grover related tutorials, [06_grover.ipynb](https://github.com/Qiskit/qiskit-tutorials/blob/master/tutorials/algorithms/06_grover.ipynb) and [07_grover_examples.ipynb](https://github.com/Qiskit/qiskit-tutorials/blob/master/tutorials/algorithms/07_grover_examples.ipynb), to use `Sampler`.

close #1365

### CODE REVIEW

**Review by dlasecki — APPROVED** [review]

**Review by woodsp-ibm — APPROVED** [review]

**Comment by HuangJunye** [comment]
@Guillermo-Mijares-Vilarino Can you please review this PR?

File: tutorials/algorithms/07_grover_examples.ipynb [hunk 1]
```diff
@@ -11,17 +11,17 @@
...
-    "from qiskit import Aer\n",
-    "from qiskit.utils import QuantumInstance\n",
-    "from qiskit.tools.visualization import plot_histogram\n",
+    "import pylab\n",
+    "\n",
     "from qiskit.algorithms import Grover, AmplificationProblem\n",
-    "from qiskit.circuit.library.phase_oracle import PhaseOracle"
+    "from qiskit.circuit.library.phase_oracle import PhaseOracle\n",
+    "from qiskit.primitives import Sampler\n",
+    "from qiskit.tools.visualization import plot_histogram"
```
Comment (Guillermo-Mijares-Vilarino) [inline tutorials/algorithms/07_grover_examples.ipynb:24]: I'd recommend to have the imports in the cell they are used. That way a user can understand better why they are importing and where it's needed.

Comment (a-matsuo) [reply tutorials/algorithms/07_grover_examples.ipynb]: Thank you for the comment. I moved the imports in the cells they are used.

**Comment by Guillermo-Mijares-Vilarino** [comment]
```
# now we can have a look at the Grover operator that is used in running the algorithm
 # (Algorithm circuits are wrapped in a gate to be appear in composition as a block
# so we have to decompose() the op to see it expanded into its component gates.)
problem.grover_operator.decompose().draw(output='mpl')
```
I think there might be a typo here. "... gate to be appear in composition..."

File: tutorials/algorithms/06_grover.ipynb [hunk 2]
```diff
@@ -292,12 +291,12 @@
...
       "text/plain": [
-       "<Figure size 267.197x204.68 with 1 Axes>"
+       "<Figure size 371.107x284.278 with 1 Axes>"
       ]
      },
-     "execution_count": 6,
+     "execution_count": 46,
      "metadata": {},
      "output_type": "execute_result"
     }
```
Comment (Guillermo-Mijares-Vilarino) [inline tutorials/algorithms/06_grover.ipynb:302]: I believe here the oracle can be simplified to only a [CCZGate](https://qiskit.org/documentation/stubs/qiskit.circuit.library.CCZGate.html#qiskit.circuit.library.CCZGate) with the [ccz method](https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.ccz.html).
```
oracle = QuantumCircuit(3)
oracle.ccz(0,1,2)
```

File: tutorials/algorithms/06_grover.ipynb [hunk 3]
```diff
@@ -371,19 +370,19 @@
...
       "text/plain": [
-       "<Figure size 146.797x325.08 with 1 Axes>"
+       "<Figure size 203.885x451.5 with 1 Axes>"
       ]
      },
-     "execution_count": 8,
+     "execution_count": 48,
      "metadata": {},
      "output_type": "execute_result"
     }
```
Comment (Guillermo-Mijares-Vilarino) [inline tutorials/algorithms/06_grover.ipynb:388]: Similarly, here `oracle.ccz(0, 1, 2)` could be a bit more straightforward to use than `oracle.append(ZGate().control(2), [0, 1, 2])`

**Comment by a-matsuo** [comment]
Thank you! I updated the notebook.

**Review by HuangJunye — APPROVED** [review]
LGTM!

**Pull request merged**